Src/Support/Google.Apis.Auth/OAuth2/ServiceAccountCredential.cs

```diff
@@ -344,7 +362,10 @@ private Task<string> GetOrCreateJwtAccessTokenAsync(string authUri)
                 }
                 // Create a new JWT.
                 var expiryUtc = nowUtc + JwtLifetime;
-                var jwtTask = Task.Run(() => CreateJwtAccessToken(authUri, nowUtc, expiryUtc));
+                Task<string> jwtTask = Task.Run(() =>
+                    HasExplicitScopes
+                    ? CreateJwtAccessTokenWithScopes(Scopes, nowUtc, expiryUtc)
+                    : CreateJwtAccessTokenWithAudience(authUri, nowUtc, expiryUtc));
```

Comment:
I think this change can be reverted and just ask for HasExplicitScopes in CreateJwtAccessToken. We don't need to disticnt methods. See my suggestion below.

Reply:
This helps to keep a single copy of the Jwt/AuthUri decision logic. Please see below comment for more detail.

Reply:
Replying below.